### ISSUE TITLE
Use standard hashCode() to identify ValueObjects in Sets instead of Javers ObjectHasher (enabled by a config param)

### ISSUE THREAD

**Issue by tkupka** [javers/javers#1301]
When you are using ValueObject with a `List` then changes to the list objects are compared by index. This allows you to detect changes on  properties of individual object(s). This is not the case for `Set/Collection` because those are not ordered containers. Here it's impossible to detect changes between the same objects on left/right side. Normally in JAVA `hashCode()` method is used to identify the same objects. But in `ObjectHasher` the object is serialized to JSON and then a hash is calculated. It will always produce different hash even JAVA's `hashCode()` produce the same hash for left/right object. As a result if you changes and object's property  inside a collection it will be identified as different object and all props are threated as changed.
With a hack to `ObjectHasher` it's working fine

```
String hash(List<LiveCdo> objects) {
		String hashes = objects.stream()
			.map(cdo -> String.valueOf(cdo.getWrappedCdo().orElse(cdo).hashCode()))
			.collect(Collectors.joining());
		return ShaDigest.longDigest(hashes);
	}
```

**Comment by bartoszwalacik** [comment]
Comparing ValueObjects by using its property values (and not by equals()) is deliberated here and consistent with other modules of the  diff algorithm. I don't see good reason to change this default behavior

**Comment by tkupka** [comment]
I understand that it's coherent with other modules. But how to detect changes on a `Collection/Set` of Value Objects  ? Can you recommend me a way how to achieve that ? Let's say that I have value object which has a reference to a collection of another value objects. If I change one property form one collection object I'll expect that `diff` should result with the info that one property (of particular object) has been changed not complete collection.
Thx for suggestion.

**Comment by bartoszwalacik** [comment]
well, the problem is that Javers wants to be **sure** that a hash of ValueObject is value based, which is not true for standard `Object.hashCode()`. I agree that it would be nice to force Javers, in some cases, to use your `hashCode()`, if you are sure that it's value based. But it can't be default, because in many cases, people don't bother to override `hashCode()` in their classes. They shouldn't. So we come to the conclusion that a switch is needed. 
So you have 2 options: contribute the feature as you suggested, but behind a switch, or map your Objects stored in Sets as Entities

**Comment by tkupka** [comment]
Hello,
thanks for considering this ticket as useful feature. I agree that It's not so good idea to make it default since in you use the different alg. for comparison. I'll take a look to the code and contribute with this feature.

**Comment by tkupka** [comment]
Hello,
I have prepared the change but I'm unable to create a branch

**Comment by bartoszwalacik** [comment]
Hi, you need to fork this repo

**Comment by tkupka** [comment]
Hi, 
Can you please check the PR if it's fine for merging?

**Comment by tkupka** [comment]
Groovy test created and other is little bit improved

**Comment by bartoszwalacik** [comment]
A new method added to `JaversBuilder`
```
    public JaversBuilder registerObjectHasher(Class<? extends ObjectHasher> objectHasherType) {
    	objectHasherImplementation = objectHasherType;
    	return this;
    }
```

**Comment by bartoszwalacik** [comment]
fixed in 7.4.0
